lib/salve/patterns.js

```diff
@@ -767,9 +764,9 @@ function PartialMatch(length) {
  * reports whether the structure represented by these events is
  * valid.
  *
- * This base class records only keeps a minimal number of properties
+ * This base class record only keeps a minimal number of properties
```

Comment:
There's no thing named a "base class record".

What you had there was a combination of two sentences:

> This base class records only a minimal number...

and 

> This base class only keeps a minimal number...

Ultimately, I prefer "records" here so go with the first quote above.